### PR TITLE
Fixes #2172. Fix error expectations in static_analysis_extension_types_A02_t02.dart

### LINKED ISSUE #2172: co19/LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t02
The representation is a final field, it cannot be assigned in a constructor body.

```
FAILED: dart2analyzer-none release_arm64 co19/LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t02
Expected: Pass
Actual: MissingCompileTimeError

--- Command "dart2analyzer" (took 07.000282s):
DART_CONFIGURATION=ReleaseARM64 sdk/bin/dart --packages=.dart_tool/package_config.json pkg/analyzer_cli/bin/analyzer.dart --use-analysis-driver-memory-byte-store --dart-sdk=/Users/scheglov/Source/Dart/sdk.git/sdk/sdk -Dtest_runner.configuration=custom-configuration-1 --enable-experiment=inline-class --ignore-unrecognized-flags --packages=/Users/scheglov/Source/Dart/sdk.git/sdk/.dart_tool/package_config.json --format=json /Users/scheglov/Source/Dart/sdk.git/sdk/tests/co19/src/LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t02.dart

static error failures:
- Unexpected error at line 19, column 5, length 2: COMPILE_TIME_ERROR.ASSIGNMENT_TO_FINAL

- Unexpected error at line 26, column 10, length 2: COMPILE_TIME_ERROR.ASSIGNMENT_TO_FINAL

--- Re-run this test:
python3 tools/test.py -m release -c dart2analyzer -a arm64 co19/LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t02
```

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t02.dart [hunk 1]
```diff
@@ -11,21 +11,23 @@
...
 // [cfe] unspecified
     id = 0;
+//  ^^
+// [analyzer] unspecified
+// [cfe] unspecified
   }
 
-  const ET1.n1(int id) {
-//      ^^^
+   const ET1.n1(int id);
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t02.dart:25]: Spurious indentation?

**Review by eernstg — COMMENTED** [review]
I think there's a non-standard indentation in one line, otherwise fine!

**Comment by sgrekhov** [comment]
Indentation fixed

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**